tests/contracts/solidity/RandomnessLotteryDemo.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.0;
+
+import "../../../precompiles/randomness/Randomness.sol";
+import "../../../precompiles/randomness/RandomnessConsumer.sol";
+
+/// @notice Smart contract to demonstrate how to use requestLocalVRFRandomWords
+contract RandomnessLotteryDemo is RandomnessConsumer {
+    /// @notice The Randomness Precompile Interface
+    Randomness public randomness =
+        Randomness(0x0000000000000000000000000000000000000809);
+
+    /// @notice The lottery has requested random words and is waiting for fulfillment
+    error WaitingFulfillment();
+
+    /// @notice The lottery doesn't have enough participants to start
+    error NotEnoughParticipants(uint256 value, uint256 required);
+
+    /// @notice The lottery doesn't accept additional participants
+    error TooManyParticipants(uint256 value, uint256 required);
+
+    /// @notice There are not enough fee to start the lottery
+    error NotEnoughFee(uint256 value, uint256 required);
+
+    /// @notice The provided fee to participate doesn't match the required amount
+    error InvalidParticipationFee(uint256 value, uint256 required);
+
+    /// @notice Event sent when a winner is awarded
+    /// @param winner The participant getting awarded
+    /// @param randomWord The randomWord being used (for informative purposes)
+    /// @param amount The amount being awarded
+    event Awarded(address indexed winner, uint256 randomWord, uint256 amount);
```

Comment:
It might be useful to know the number of participants with this event

Reply:
I added more events with those numbers